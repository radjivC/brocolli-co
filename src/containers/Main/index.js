import React, { Component } from 'react';
import Modal from 'react-modal'
import Subscribe from '../../components/Subscribe'
import Header from '../../components/Header'
import Footer from '../../components/Footer'

import 'bulma/css/bulma.css'


const customStyles = {
  content: {
    top: '50%',
    height: '90%',
    width: '90%',
    maxWidth: '430px',
    maxHeight: '400px',
    left: '50%',
    right: 'auto',
    bottom: 'auto',
    marginRight: '-50%',
    transform: 'translate(-50%, -50%)'
  }
}


export default class App extends Component {

  constructor(props) {
   super(props)

   this.state = {
     modalIsOpen: false
   }
 }

  openModal = () => {
    this.setState({ modalIsOpen: true })
  }

  closeModal = () => {
    this.setState({ modalIsOpen: false })
  }

  render() {
    return (
      <div>
        <section className="hero is-info is-fullheight">
          <Header />
          <div className="hero-body">
            <div className="container has-text-centered">
              <div className="column is-6 is-offset-3">
                <h1 className="title">
                  A better way to enjoy every day.
                </h1>
                <h2 className="subtitle">
                   Be the first to know when we launch
                </h2>
                  <a className="bd-tw-button button is-danger" onClick={this.openModal} >
                    <span className="icon">
                      <i className="fa fa-paper-plane" />
                    </span>
                    <span className="is-capitalized">Request an invite</span>
                  </a>

                  <Modal
                    isOpen={this.state.modalIsOpen}
                    onAfterOpen={this.afterOpenModal}
                    onRequestClose={this.closeModal}
                    style={customStyles}
                    ariaHideApp={false}
                  >
                    <span className="icon is-pulled-right" onClick={this.closeModal}>
                      <i className="fa fa-window-close" aria-hidden="true" />
                    </span>
                    <Subscribe closeModalRequest={this.closeModal} />
                  </Modal>
                  </div>
                </div>
              </div>

              <Footer />
        </section>
      </div>
    );
  }
}
