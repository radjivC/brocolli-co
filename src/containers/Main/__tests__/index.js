import React from 'react'
import { mount } from 'enzyme'
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'
import Main from '../'

configure({ adapter: new Adapter() })

describe('Subscribe', () => {
  it('can render without crashing', () => {
    const component = renderer.create(
      <Main />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('can render done', () => {
    const wrapper = mount(<Main />)
    wrapper.setState({ modalIsOpen:false })
    expect(wrapper.find('Header').length).toEqual(1)
    expect(wrapper.find('Footer').length).toEqual(1)
    expect(wrapper.find('Modal').length).toEqual(1)
    expect(wrapper.find('a.bd-tw-button.button.is-danger').text()).toEqual('Request an invite')
  })

  it('can render the modal', () => {
    const wrapper = mount(<Main />)
    wrapper.setState({ modalIsOpen:true })
    expect(wrapper.find('Header').length).toEqual(1)
    expect(wrapper.find('Footer').length).toEqual(1)
    expect(wrapper.find('Modal').length).toEqual(1)
    expect(wrapper.find('Subscribe').length).toEqual(1)
  })
})
