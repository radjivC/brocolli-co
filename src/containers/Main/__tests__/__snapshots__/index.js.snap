// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Subscribe can render without crashing 1`] = `
<div>
  <section
    className="hero is-info is-fullheight"
  >
    <div
      className="hero-head"
    >
      <nav
        className="navbar"
      >
        <div
          className="container"
        >
          <div
            className="navbar-brand"
          >
            <a
              className="navbar-item"
            >
              Brocoli and Co
            </a>
          </div>
        </div>
      </nav>
    </div>
    <div
      className="hero-body"
    >
      <div
        className="container has-text-centered"
      >
        <div
          className="column is-6 is-offset-3"
        >
          <h1
            className="title"
          >
            A better way to enjoy every day.
          </h1>
          <h2
            className="subtitle"
          >
            Be the first to know when we launch
          </h2>
          <a
            className="bd-tw-button button is-danger"
            onClick={[Function]}
          >
            <span
              className="icon"
            >
              <i
                className="fa fa-paper-plane"
              />
            </span>
            <span
              className="is-capitalized"
            >
              Request an invite
            </span>
          </a>
        </div>
      </div>
    </div>
    <footer
      className="footer"
    >
      <div
        className="container"
      >
        <div
          className="content has-text-centered"
        >
          <p
            className="has-text-primary"
          >
            <strong>
              Build in Melbourne
            </strong>
          </p>
        </div>
      </div>
    </footer>
  </section>
</div>
`;
